jscomp/ml/typecore.ml

```diff
@@ -2100,7 +2100,12 @@ and type_expect_ ?in_function ?(recarg=Rejected) env sexp ty_expected =
         let lbl_exp_list =
           wrap_disambiguate "This record expression is expected to have" ty_record
             (type_label_a_list loc true env
-               (fun e k -> k (type_label_exp true env loc ty_record e))
+               (fun (id, ld, exp) k ->
+                let exp =
+                  if ld.lbl_repres = Record_object && is_option_type env ld.lbl_arg then
+                    {exp with pexp_desc = Pexp_construct ({id with txt = Longident.Lident "Some"}, Some exp)}
+                  else exp in
```

Comment:
Maybe can we check the exp whether is option type, then `option<'a>` -> `option<'a>` else `'a` -> `option<'a>`? Therefore we can avoid the `option<option<int>>`

Reply:
Yes this is just testing that the types are right. The option-of-option mechanism is in the compiler already and does the same thing.
The example just checks that it kicks in as expected.

Reply:
More precisely. We know the type checker does the right thing. And the test with option of option tests that the code generation also does the right thing.
Without option of option, it would be hard to say, as `3` is the same as `Some(3)` in terms of the code which is generated.

Reply:
Totally agreed.